starknet_py/tests/e2e/account/outside_execution_test.py

```diff
@@ -0,0 +1,136 @@
+
+
+@pytest.mark.asyncio
+async def test_account_outside_execution_any_caller(
+    argent_account: BaseAccount,
+    map_contract,
+):
+
+    assert any(
+        [
+            await argent_account.supports_interface(OutsideExecutionInterfaceID.V1),
+            await argent_account.supports_interface(OutsideExecutionInterfaceID.V2),
+        ]
+    )
+
+    put_call = Call(
+        to_addr=map_contract.address,
+        selector=get_selector_from_name("put"),
+        calldata=[20, 20],
+    )
+
+    call = await argent_account.sign_outside_execution_call(
+        calls=[
+            put_call,
+            put_call,
+            put_call,
+        ],
+        execution_time_bounds=OutsideExecutionTimeBounds(
+            execute_after=datetime.datetime.now() - datetime.timedelta(hours=1),
+            execute_before=datetime.datetime.now() + datetime.timedelta(hours=1),
+        ),
+        caller=ANY_CALLER,
+    )
+
+    tx = await argent_account.execute_v1(calls=[call], max_fee=MAX_FEE)
+    await argent_account.client.wait_for_tx(tx.transaction_hash)
+
+
+@pytest.mark.asyncio
+async def test_account_outside_execution_for_invalid_caller(
+    argent_account: BaseAccount,
+    account: BaseAccount,
+    map_contract,
+):
+    assert any(
+        [
+            await argent_account.supports_interface(OutsideExecutionInterfaceID.V1),
+            await argent_account.supports_interface(OutsideExecutionInterfaceID.V2),
+        ]
+    )
+
+    put_call = Call(
+        to_addr=map_contract.address,
+        selector=get_selector_from_name("put"),
+        calldata=[20, 20],
+    )
+
+    call = await argent_account.sign_outside_execution_call(
+        calls=[
+            put_call,
+            put_call,
+            put_call,
+        ],
+        execution_time_bounds=OutsideExecutionTimeBounds(
+            execute_after=datetime.datetime.now() - datetime.timedelta(hours=1),
+            execute_before=datetime.datetime.now() + datetime.timedelta(hours=1),
+        ),
+        caller=account.address,
+    )
+
+    tx = await argent_account.execute_v1(calls=[call], max_fee=MAX_FEE)
+
+    with pytest.raises(TransactionRevertedError) as err:
+        await argent_account.client.wait_for_tx(tx.transaction_hash)
+
+    assert "argent/invalid-caller" in err.value.message
+
+
+@pytest.mark.asyncio
+async def test_account_outside_execution_for_impossible_timebounds(
+    argent_account: BaseAccount,
+    map_contract,
+):
+
+    assert any(
+        [
+            await argent_account.supports_interface(OutsideExecutionInterfaceID.V1),
+            await argent_account.supports_interface(OutsideExecutionInterfaceID.V2),
+        ]
+    )
+
+    put_call = Call(
+        to_addr=map_contract.address,
+        selector=get_selector_from_name("put"),
+        calldata=[20, 20],
+    )
+
+    call = await argent_account.sign_outside_execution_call(
+        calls=[
+            put_call,
+            put_call,
+            put_call,
+        ],
```

Comment:
Why 3 times 😅 ?

Reply:
Felt like it's a slightly better test case.